recipes/cppcheck/all/conanfile.py

```diff
@@ -29,7 +29,7 @@ def config_options(self):
         if Version(self.version) >= Version("2.8.0"):
             del self.options.with_z3
 
-    def requirements(self):
+    def build_requirements(self):
```

Comment:
It doesn't make sense. If you are trying to emulate private dependencies, I can already say you are losing your time, it will work in conan v2 only.